### ISSUE TITLE
Deploy your website

### ISSUE THREAD

**Issue by nidalaa** [cool-beetles/my-first-project#1]
Let's make your work available in internet!

In order to deploy your application:
- create an account in https://www.heroku.com/
- make sure that your repo has installed Git (of course ;) ) and `heroku cli ` https://devcenter.heroku.com/articles/git#prerequisites-installing-git-and-the-heroku-cli
- create a heroku remote (which is like a copy of your git repo stored in heroku) https://devcenter.heroku.com/articles/git#creating-a-heroku-remote, it will automatically create a web application in your heroku account https://dashboard.heroku.com/apps
- after that every time you push to Github your code is just stored there as it used to be but if you want to make your changes live in your app published in Internet you need to `deploy` your changes (which means push to heroku repo) https://devcenter.heroku.com/articles/git#deploying-code

-------------------
In short:
`git push master` ----> pushing changes to Github repo
`git push heroku master` ----> pushing changes to heroku repo = deploy your changes to we app
-------------------

**Comment by nidalaa** [comment]
EDIT: Heroku does not serve static html pages.

That's why we need to turn this app to simple Ruby + Rack app. Here is a full article: https://devcenter.heroku.com/articles/static-sites-ruby

And here is a shortcut for your app:

- instead of https://devcenter.heroku.com/articles/static-sites-ruby#create-directory-structure run

```
mkdir -p public/{images,js,css}
touch {config.ru,public/index.html}
bundle init
```

- then go with https://devcenter.heroku.com/articles/static-sites-ruby#specify-dependencies
- instead of https://devcenter.heroku.com/articles/static-sites-ruby#file-server-configuration put this in your config.ru file:

```
use Rack::Static,
  :urls => ["/images", "/js", "/css"],
  :root => "public"

run lambda { |env|
  path = env['REQUEST_PATH'] != "/" ? env['REQUEST_PATH'] : "/home.html"
  [
    200,
    {
      'Content-Type'  => 'text/html',
      'Cache-Control' => 'public, max-age=86400'
    },
    File.open("public#{path}", File::RDONLY)
  ]
}

```


- and then move files:

```
mv *.html public/
```

```
mv style.css public/css/
```

**and change references from `style.css` to `css/style.css`**

```
mv WebDev_files/* /public/images/
```

**and change ferences from `WebDev_files/` to `images/` in all files which include images**

- then run your rack server locally as in https://devcenter.heroku.com/articles/static-sites-ruby#running-locally ang check your app by writing `http://localhost:9292` in your web browser.

Congratulations! You've just written a rack server which serves your html files :D
Try to understand what is going on, check if everything works well (some image references might need additional editions) and try to deploy your app again!

---------------

After it is done please go with instructions from initial description. If heroku does not recognize ruby app you can set it up manually by running: `heroku buildpacks:set heroku/ruby`

Good luck! :D

**Comment by nidalaa** [comment]
@cool-beetles could you add a link to your app in Readme?

**Comment by nidalaa** [comment]
Awesome! 😍
